### PR TITLE
Forward custom headers

### LINKED ISSUE #129: add support for custom Host headers when sending requests to a local server
### Problem
When trying to use **stripe-cli** together with a server behind **puma-dev** I get errors as the Go HTTP Client doesn't resolve `.test` domains to `http://127.0.0.1:80`.

### Feature
Have a flag to set a custom `Host` Header on requests.

### Examples

`$ stripe listen --forward-to http://127.0.0.1:80 --forward-to-host foo.test`

### WIP implementation

https://github.com/stripe/stripe-cli/compare/master...romainmenke:master

### PR DESCRIPTION
**Pull request by mimi-stripe**
### Reviewers
r? @joek-stripe @joek
cc @stripe/dev-platform @ob-stripe @ob

 ### Summary
Allows users to add custom headers when running CLI "listen" command. Strips whitespace and sanitizes sub 0x20 ascii characters.

Fixes #129.

### CODE REVIEW

File: pkg/cmd/listen.go [hunk 1]
```diff
@@ -65,6 +67,8 @@ to your localhost:
 
 	lc.cmd.Flags().StringSliceVarP(&lc.events, "events", "e", []string{"*"}, "A comma-separated list of which webhook events to listen for. For a list of all possible events, see: https://stripe.com/docs/api/events/types")
 	lc.cmd.Flags().StringVarP(&lc.forwardURL, "forward-to", "f", "", "The URL to forward webhook events to")
+	lc.cmd.Flags().StringSliceVarP(&lc.forwardHeaders, "headers", "H", []string{}, "A comma-separated list of custom headers to forward.")
```
Comment (ob-stripe) [inline pkg/cmd/listen.go]: Nit, for consistency with other help messages:
```suggestion
	lc.cmd.Flags().StringSliceVarP(&lc.forwardHeaders, "headers", "H", []string{}, "A comma-separated list of custom headers to forward")
```

File: pkg/cmd/listen.go [hunk 2]
```diff
@@ -65,6 +67,8 @@ to your localhost:
...
+	lc.cmd.Flags().StringSliceVar(&lc.forwardConnectHeaders, "connect-headers", []string{}, "A comma-separated list of custom headers to forward for connect.")
```
Comment (ob-stripe) [inline pkg/cmd/listen.go]: More nits:
```suggestion
	lc.cmd.Flags().StringSliceVar(&lc.forwardConnectHeaders, "connect-headers", []string{}, "A comma-separated list of custom headers to forward for Connect")
```

File: pkg/cmd/listen.go [hunk 3]
```diff
@@ -85,6 +89,9 @@ to your localhost:
 // Normally, this function would be listed alphabetically with the others declared in this file,
 // but since it's acting as the core functionality for the cmd above, I'm keeping it close.
 func (lc *listenCmd) runListenCmd(cmd *cobra.Command, args []string) error {
+	
+	print("running listen command")
```
Comment (ob-stripe) [inline pkg/cmd/listen.go]: I assume this is a leftover debug statement?

File: pkg/cmd/listen.go [hunk 4]
```diff
@@ -172,7 +190,9 @@ func (lc *listenCmd) getEndpointsFromAPI(secretKey string) requests.WebhookEndpo
 	return examples.WebhookEndpointsList()
 }
 
-func buildEndpointRoutes(endpoints requests.WebhookEndpointList, forwardURL, forwardConnectURL string) []proxy.EndpointRoute {
+// add custom host as arg
+// TODO: check if forward headers exist
```
Comment (ob-stripe) [inline pkg/cmd/listen.go]: Is this comment still applicable?

File: pkg/proxy/endpoint.go [hunk 5]
```diff
@@ -128,6 +143,7 @@ func NewEndpointClient(url string, connect bool, events []string, cfg *EndpointC
 
 	return &EndpointClient{
 		URL:     url,
+		headers: convertToMapAndSanitize(headers),
```
Comment (ob-stripe) [inline pkg/proxy/endpoint.go]: 💯

File: pkg/proxy/endpoint.go [hunk 6]
```diff
@@ -154,3 +170,29 @@ func convertToMap(events []string) map[string]bool {
...
+		// what to do if regex fails to compile?
+		fmt.Println("error with regex")
+		return headerMap
+	}
+
+	for _, header := range headers {
+
+		header = reg.ReplaceAllString(header, "")
+
+		splitHeader := strings.Split(header, ":")
```
Comment (ob-stripe) [inline pkg/proxy/endpoint.go]:
```suggestion
		splitHeader := strings.SplitN(header, ":", 2)
```

This way, the value can contain `:` characters.

File: pkg/cmd/listen.go (hunk 4 above)
Comment (mimi-stripe) [reply pkg/cmd/listen.go]: Nope! I'll remove it!

File: pkg/proxy/endpoint.go (hunk 6 above)
Comment (mimi-stripe) [reply pkg/proxy/endpoint.go]: Good idea!

File: pkg/proxy/endpoint.go [hunk 7]
```diff
@@ -154,3 +170,29 @@ func convertToMap(events []string) map[string]bool {
...
 	return eventsMap
 }
+
+func convertToMapAndSanitize(headers []string) map[string]string {
+	reg, err := regexp.Compile("[\x00-\x1f]+")
+
+	headerMap := make(map[string]string)
+
+	if err != nil {
+		// what to do if regex fails to compile?
```
Comment (joek-stripe) [inline pkg/proxy/endpoint.go]: `panic(err)` https://gobyexample.com/panic

Comment (mimi-stripe) [reply pkg/proxy/endpoint.go]: So turns out I can use "regexp.MustCompile" which automatically panics instead of returning an error!

**Pull request merged**